miriway.cpp

```diff
@@ -131,6 +132,21 @@ struct ShellPids
     std::mutex mutable shell_component_mutex;
     std::set<pid_t> shell_component_pids;
 };
+
+auto to_key(std::string_view as_string) -> xkb_keysym_t
```

Comment:
Maybe this could be (untested):
```
auto keysym = xkb_keysym_from_name(std::format("XKB_KEY_{}", as_string).c_str(), 0);
if (keysym == XKB_KEY_NoSymbol)
{
    keysym = xkb_keysym_from_name(std::format("XKB_KEY_{}", as_string).c_str(), XKB_KEYSYM_CASE_INSENSITIVE);
    // Maybe warn if the case-insensitive search succeeded
}
if (keysym == XKB_KEY_NoSymbol)
{
    throw mir::AbnormalExit((std::format("Unrecognised key in config: {}", as_string).c_str());
}
return keysym
```

Reply:
Good thought. And if I didn't shorten the strings in the config and required `XKB_KEY_foo` in full it would be even simpler.

Reply:
Actually, the `XKB_KEY_` prefix shouldn't be there. So that was even simpler!